src/physics/utils/physics_data.F90

```diff
@@ -501,36 +577,54 @@ subroutine check_field_3d(file, var_names, vcoord_name, timestep,          &
 
    end subroutine check_field_3d
 
-   subroutine write_check_field_entry(stdname, diff_count, max_diff, is_first)
+   subroutine write_check_field_entry(stdname,      diff_count,               &
+                                      max_diff,     max_diff_rank,            &
+                                      max_diff_col, is_first,                 &
+                                      max_diff_lev)
 
       use cam_logfile, only: iulog
+      use shr_kind_mod, only: cs=>shr_kind_cs
 
       !Dummy variables:
-      character(len=*), intent(in) :: stdname
-      integer,          intent(in) :: diff_count
-      real(kind_phys),  intent(in) :: max_diff
-      logical,          intent(in) :: is_first
+      character(len=*),  intent(in) :: stdname
+      integer,           intent(in) :: diff_count
+      real(kind_phys),   intent(in) :: max_diff
+      integer,           intent(in) :: max_diff_rank !MPI rank max diff occurred on
+      integer,           intent(in) :: max_diff_col  !max diff column (1st) dimension value
+      integer, optional, intent(in) :: max_diff_lev  !max diff level (2nd) dimension value
+      logical,           intent(in) :: is_first
 
       !Local variables:
-      character(len=24)            :: fmt_str
+      character(len=cs)            :: fmt_str
+      character(len=cs)            :: index_str
       integer                      :: slen
       integer, parameter           :: indent_level = 50
 
       slen = len_trim(stdname)
 
+      !Write check_field log header:
       if (is_first) then
          write(iulog, *) ''
-         write(fmt_str, '(a,i0,a)') "(1x,a,t",indent_level+1,",1x,a,2x,a)"
-         write(iulog, fmt_str) 'Variable', '# Diffs', 'Max Diff'
-         write(iulog, fmt_str) '--------', '-------', '--------'
+         write(fmt_str, '(a,i0,a)') "(1x,a,t",indent_level+1,",1x,a,2x,a,3x,a)"
+         write(iulog, fmt_str) 'Variable', '# Diffs', 'Max Diff', 'Max Diff loc (rank, col, lev)'
+         write(iulog, fmt_str) '--------', '-------', '--------', '-----------------------------'
       end if
 
+      !Write standard name separately if longer
+      !than the indent level:
       if (slen > indent_level) then
          write(iulog, '(a)') trim(stdname)
          slen = 0
       end if
-      write(fmt_str, '(a,i0,a)') "(1x,a,t",indent_level+1,",1x,i7,2x,e8.2)"
-      write(iulog, fmt_str) stdname(1:slen), diff_count, max_diff
+
+      !Write out difference and index valuesa:
+      if (present(max_diff_lev)) then
+         write(index_str, '(a,i0,a,i0,a,i0,a)') "(",max_diff_rank,",",max_diff_col,",",max_diff_lev,"}"
```

Comment:
why the different types of parenthesis/brackets?

Reply:
There are two possibilities.  One is that I have a deep, secret, subconscious hatred of parenthetical symmetry.  The other is that I made a typo and then copied that typo.  I'll leave it up to you to decide which is the most likely option.

Also, parenthesis are now both the same.